jmxtrans-output/jmxtrans-output-core/src/test/java/com/googlecode/jmxtrans/model/output/GraphiteWriterTests.java

```diff
@@ -121,6 +121,20 @@ public void allowDottedWorks() throws Exception {
 	}
 
 	@Test
+	public void useAllTypeNamesWorks() throws Exception {
+		Server server = Server.builder().setHost("host").setPort("123").setAlias("host").build();
+		Query query = Query.builder().setUseAllTypeNames(true).build();
+		String typeName = "typeName,typeNameKey1=typeNameValue1,typeNameKey2=typeNameValue2";
+		Result result = new Result(System.currentTimeMillis(), "attributeName", "className", "objDomain", null, typeName, ImmutableMap.of("key", (Object)1));
+
+		ByteArrayOutputStream out = new ByteArrayOutputStream();
+		GraphiteWriter writer = getGraphiteWriter(out);
+		writer.doWrite(server, query, of(result));
+
+		Assertions.assertThat( out.toString()).startsWith("servers.host.className.typeNameValue1_typeNameValue2.attributeName_key 1 ");
```

Comment:
How do you ensure that there is a consistent ordering of the values ? In your implementation, `typeName="typeName,typeNameKey1=typeNameValue1,typeNameKey2=typeNameValue2"` will produce a different output than `typeName="typeName,typeNameKey2=typeNameValue2,typeNameKey1=typeNameValue1"` enven if they are functionally equal. I would need to check , but I think that the JMX spec does not specify an ordering of the keys. So that's probable a test you shoudl add and make sure it passes.